src/strat_engine/engine.rs

```diff
@@ -45,9 +47,30 @@ impl Engine for StratEngine {
             return Err(EngineError::Stratis(ErrorEnum::AlreadyExists(name.into())));
         }
 
-        let pool_uuid = Uuid::new_v4();
-        let bds = try!(BlockDev::initialize(&pool_uuid, blockdev_paths, MIN_MDA_SIZE, true));
-        let pool = StratPool::new(name, pool_uuid, &bds, raid_level);
+        let mut devices = BTreeSet::new();
+        for path in blockdev_paths {
+            let dev = try!(Device::from_str(&path.to_string_lossy()));
+            devices.insert(dev);
+        }
+
+        if devices.len() != blockdev_paths.len() {
```

Comment:
I'm afraid I must disagree. If the user asks for a pool comprised of three blockdevs, we either need to give them a pool of three blockdevs, or fail.